src/master_element/ngp/hex8/Hex8SCS.C

```diff
@@ -0,0 +1,144 @@
+//-------- hex8_derivative -------------------------------------------------
+//--------------------------------------------------------------------------
+void Hex8SCS::hex8_derivative(
+  const int npts,
+  SharedMemView<DoubleType**> &intgLoc,
+  SharedMemView<DoubleType***> &deriv)
+{
+  const double half = 0.50;
+  const double one4th = 0.25;
+  for (int  j = 0; j < npts; ++j) {
+
+    const double s1 = intgLoc_(j,0);
+    const double s2 = intgLoc_(j,1);
+    const double s3 = intgLoc_(j,2);
+    const double s1s2 = s1*s2;
+    const double s2s3 = s2*s3;
+    const double s1s3 = s1*s3;
+
+    // shape function derivative in the s1 direction -
+    deriv(j,0,0) = half*( s3 + s2 ) - s2s3 - one4th;
+    deriv(j,1,0) = half*(-s3 - s2 ) + s2s3 + one4th;
+    deriv(j,2,0) = half*(-s3 + s2 ) - s2s3 + one4th;
+    deriv(j,3,0) = half*(+s3 - s2 ) + s2s3 - one4th;
+    deriv(j,4,0) = half*(-s3 + s2 ) + s2s3 - one4th;
+    deriv(j,5,0) = half*(+s3 - s2 ) - s2s3 + one4th;
+    deriv(j,6,0) = half*(+s3 + s2 ) + s2s3 + one4th;
+    deriv(j,7,0) = half*(-s3 - s2 ) - s2s3 - one4th;
+
+    // shape function derivative in the s2 direction -
+    deriv(j,0,1) = half*( s3 + s1 ) - s1s3 - one4th;
+    deriv(j,1,1) = half*( s3 - s1 ) + s1s3 - one4th;
+    deriv(j,2,1) = half*(-s3 + s1 ) - s1s3 + one4th;
+    deriv(j,3,1) = half*(-s3 - s1 ) + s1s3 + one4th;
+    deriv(j,4,1) = half*(-s3 + s1 ) + s1s3 - one4th;
+    deriv(j,5,1) = half*(-s3 - s1 ) - s1s3 - one4th;
+    deriv(j,6,1) = half*( s3 + s1 ) + s1s3 + one4th;
+    deriv(j,7,1) = half*( s3 - s1 ) - s1s3 + one4th;
+    
+    // shape function derivative in the s3 direction -
+    deriv(j,0,2) = half*( s2 + s1 ) - s1s2 - one4th;
+    deriv(j,1,2) = half*( s2 - s1 ) + s1s2 - one4th;
+    deriv(j,2,2) = half*(-s2 - s1 ) - s1s2 - one4th;
+    deriv(j,3,2) = half*(-s2 + s1 ) + s1s2 - one4th;
+    deriv(j,4,2) = half*(-s2 - s1 ) + s1s2 + one4th;
+    deriv(j,5,2) = half*(-s2 + s1 ) - s1s2 + one4th;
+    deriv(j,6,2) = half*( s2 + s1 ) + s1s2 + one4th;
+    deriv(j,7,2) = half*( s2 - s1 ) - s1s2 + one4th;
+  }
+}
+
+//--------------------------------------------------------------------------
+//-------- shape_fcn -------------------------------------------------------
+//--------------------------------------------------------------------------
+void
+Hex8SCS::shape_fcn(SharedMemView<DoubleType**> &shpfc)
+{
+  hex_shape_fcn(&numIntPoints_, &intgLoc_, shpfc);
+}
+
+//--------------------------------------------------------------------------
+//-------- hex8_shape_fcn --------------------------------------------------
+//--------------------------------------------------------------------------
+void
+Hex8SCS::hex8_shape_fcn(
+  const int  &npts,
+  const double *isoParCoord, 
+  SharedMemView<DoubleType**> &shape_fcn)
+{
+  const double one8th = 0.125;
+  const double half = 0.50;
+  for ( int j = 0; j < numIntPoints_; ++j ) {
+    
+    const double s1 = isoParCoord(j,0);
```

Comment:
isoParCoord is passed into hex8_shape_fcn as 'const double*', but appears to be indexed as a 2D view. Is that intended?